crates/client/src/serialize/txt/zone_lex.rs

```diff
@@ -156,18 +154,18 @@ impl<'a> Lexer<'a> {
                                 ))
                             })?;
 
-                            if "INCLUDE" == dollar {
-                                return Ok(Some(Token::Include));
-                            } else if "ORIGIN" == dollar {
-                                return Ok(Some(Token::Origin));
-                            } else if "TTL" == dollar {
-                                return Ok(Some(Token::Ttl));
-                            } else {
-                                return Err(LexerErrorKind::UnrecognizedDollar(
-                                    char_data.take().unwrap_or_else(|| "".into()),
-                                )
-                                .into());
-                            }
+                            let tok = match &*dollar {
```

Comment:
Style nit: instead of naming this value, just `return` here and wrap the values in `Ok` inside the match arms?

Reply:
No problem with that, just a personal habit to avoid variant construction noise.

Also would change `&*dollar` to `dollar.as_str()` more chars but more direct.

Reply:
By he way is it better?

```rust
                            return match dollar.as_str() {
                                "INCLUDE" => Ok(Some(Token::Include)),
                                "ORIGIN" => Ok(Some(Token::Origin)),
                                "TTL" => Ok(Some(Token::Ttl)),
                                _ => {
                                    Err(LexerErrorKind::UnrecognizedDollar(
                                        char_data.take().unwrap_or_else(|| "".into()),
                                    )
                                    .into())
                                }
                            };
```

Reply:
I think so! You could also keep the `Ok(Some())` on the outside and return the `Err` case from inside the match arm.